go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -275,15 +275,15 @@ func (oc *DefaultNetworkController) deleteEgressFirewall(egressFirewallObj *egre
 			break
 		}
 	}
+	// delete acls first, then dns address set that is referenced in these acls
+	if err := oc.deleteEgressFirewallRules(egressFirewallObj.Namespace); err != nil {
```

Comment:
it looks that way to me, but I guess it helps us so that we dont have to distinguish between egress firewall on local gw mode or shared. In shared the acls are only on the join switch, while on local gw they are on all worker switches. In the future we are looking to move to using only the worker switches. I guess though maybe there is an optimization we can do where we use a port group with a port on every switch like Dumitru did for multicast. Then we could just delete or add the acl to that port group, rather than to every switch.

Reply:
good point, I linked this discussion to the egress firewall refactoring story to address it later